### PR TITLE
[Changed all indentations to 4 spaces] Add Inbox page and Donation-match page

### PR DESCRIPTION
**Pull request by sherrysheng0208**

### CODE REVIEW

File: src/main/java/com/googlesps/feedon/servlets/DeclinedInboxServlet.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+public class DeclinedInboxServlet extends HttpServlet {
+
+    private Gson gson = new Gson();
+    private static final String JSON_CONTENT_TYPE = "application/json";
+    UserService userService = UserServiceFactory.getUserService();
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+
+    /**
+     * Get donationID from the declined donations and set the donation status of the current user to "declined"
+     * @param request
```
Comment (rashikaranpuria) [inline src/main/java/com/googlesps/feedon/servlets/DeclinedInboxServlet.java:27]: optional, java doc param also contains description

File: src/main/java/com/googlesps/feedon/servlets/DeclinedInboxServlet.java [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+        long id = Long.parseLong(request.getParameter("id"));
+        Key taskEntityKey = KeyFactory.createKey("Donation", id);
+
+        String email = userService.getCurrentUser().getEmail();
+        String userStatus = "status_" + email.replace('@', '_');
+
+        try {
+            // When the required donation can found by donationID, set the donation status to "declined"
+            Entity donationEntity = datastore.get(taskEntityKey);
+//            donationEntity.removeProperty(userStatus);
```
Comment (rashikaranpuria) [inline src/main/java/com/googlesps/feedon/servlets/DeclinedInboxServlet.java:43]: Remove stray comments

File: src/main/java/com/googlesps/feedon/servlets/DeclinedInboxServlet.java [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+            donationEntity.setProperty(userStatus, "declined");
+            datastore.put(donationEntity);
+        } catch (EntityNotFoundException e) {
+            e.printStackTrace();
+        }
+    }
+
+    /**
+     * Display all the declined donations of the currently logined charity on the Inbox page
```
Comment (rashikaranpuria) [inline src/main/java/com/googlesps/feedon/servlets/DeclinedInboxServlet.java:52]: typo, logged in

File: src/main/java/com/googlesps/feedon/servlets/DonationOfferServlet.java [hunk 4]
```diff
@@ -71,46 +62,46 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
...
-      return null;
+    @Nullable
+    private String getUploadedFileUrl(HttpServletRequest request, String formInputElementName) {
+        BlobstoreService blobstoreService = BlobstoreServiceFactory.getBlobstoreService();
+        Map<String, List<BlobKey>> blobs = blobstoreService.getUploads(request);
+        List<BlobKey> blobKeys = blobs.get("image");
+
+        // User submitted form without selecting a file, so we can't get a URL. (dev server)
+        if (blobKeys == null || blobKeys.isEmpty()) {
+        return null;
```
Comment (rashikaranpuria) [inline src/main/java/com/googlesps/feedon/servlets/DonationOfferServlet.java:73]: correct indentation

File: src/main/java/com/googlesps/feedon/servlets/LoginServlet.java [hunk 5]
```diff
@@ -20,103 +20,113 @@
...
-  private boolean hasRegistered(String email) {
-    if (getUserType(email).equals("")) {
-      return false;
-    } else {
-      return true;
+    /**
+     * Return whether this email address is already registered, i.e.
+     * in the datastore with "User" entity.
+     */
+    private boolean hasRegistered(String email) {
```
Comment (rashikaranpuria) [inline src/main/java/com/googlesps/feedon/servlets/LoginServlet.java:125]: Great abstraction

File: src/main/webapp/styles/inbox.css [hunk 6]
```diff
@@ -32,18 +32,32 @@ h1 {
...
+    margin-right: 30px;
     text-align: center;
+    text-decoration: none;
 }
 
 .inbox-btn:hover {
-    opacity: 1;
-}
\ No newline at end of file
+    opacity: 2;
```
Comment (rashikaranpuria) [inline src/main/webapp/styles/inbox.css:52]: I think valid opacity values are from 0 to 1 🤔

File: src/main/java/com/googlesps/feedon/servlets/AcceptedInboxServlet.java [hunk 7]
```diff
@@ -0,0 +1,138 @@
...
+import java.util.List;
+
+@WebServlet("/inbox-accepted")
+public class AcceptedInboxServlet extends HttpServlet {
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    private final String EMPTY_STRING = "";
+    UserService userService = UserServiceFactory.getUserService();
+    private Gson gson = new Gson();
+    private static final String JSON_CONTENT_TYPE = "application/json";
```
[the remark this answers is not in the document]
Comment (sherrysheng0208) [reply src/main/java/com/googlesps/feedon/servlets/AcceptedInboxServlet.java:24]: sure

**Pull request merged**